code/__DEFINES/traits.dm

```diff
@@ -388,6 +388,8 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 #define TRAIT_NODROP "nodrop"
 /// cannot be inserted in a storage.
 #define TRAIT_NO_STORAGE_INSERT "no_storage_insert"
+/// cannot be inserted into a machine which has storage (currently anything using component/material_container, experimentor and destructive analyser)
+#define TRAIT_NO_MACHINE_INSERT "no_machine_insert"
```

Comment:
Rather than depending on things checking a trait the holderloving component should be interrupting attempts to move the item, like it already does. You should make it able to interrupt being moved by material containers and destructive analyzer instead of this.

Reply:
plz mark as draft